Bottom: d6530e1c569a3ed11b2f0cd3c27bc26fae887dc9
Top:    cc627a383bca7302d2c50b2a8884ebed0e3d1b99
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 16:18:11 +0000

Refactor to use step_id, not int


---

diff --git a/src/Makefile.am b/src/Makefile.am
index 35d6178..60807b7 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -169,6 +169,7 @@ PCB_SRCS = \
 	hid/common/object3d.h \
 	hid/common/quad.c \
 	hid/common/quad.h \
+	hid/common/step_id.h \
 	hid/common/vertex3d.c \
 	hid/common/vertex3d.h \
 	hid/hidint.h 
diff --git a/src/hid/common/edge3d.c b/src/hid/common/edge3d.c
index 263f986..35a8846 100644
--- a/src/hid/common/edge3d.c
+++ b/src/hid/common/edge3d.c
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <glib.h>
 
+#include "step_id.h"
 #include "edge3d.h"
 
 edge_info *
diff --git a/src/hid/common/edge3d.h b/src/hid/common/edge3d.h
index f54468c..3080865 100644
--- a/src/hid/common/edge3d.h
+++ b/src/hid/common/edge3d.h
@@ -14,8 +14,8 @@ typedef struct
   double radius;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int infinite_line_identifier;
-  int edge_identifier;
+  step_id infinite_line_identifier;
+  step_id edge_identifier;
 } edge_info;
 
 edge_info *make_edge_info (void);
diff --git a/src/hid/common/face3d.c b/src/hid/common/face3d.c
index 0926001..5f1df0a 100644
--- a/src/hid/common/face3d.c
+++ b/src/hid/common/face3d.c
@@ -1,6 +1,7 @@
 #include <glib.h>
 #include <stdbool.h>
 
+#include "step_id.h"
 #include "quad.h"
 #include "contour3d.h"
 #include "appearance.h"
diff --git a/src/hid/common/face3d.h b/src/hid/common/face3d.h
index c630c17..1ca9eb4 100644
--- a/src/hid/common/face3d.h
+++ b/src/hid/common/face3d.h
@@ -12,9 +12,9 @@ typedef struct {
   appearance *appear;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int surface_identifier;
-  int face_identifier;
-  int face_bound_identifier;
+  step_id surface_identifier;
+  step_id face_identifier;
+  step_id face_bound_identifier;
 } face3d;
 
 face3d *make_face3d (void);
diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 0197118..34ffceb 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -5,6 +5,7 @@
 
 #include <glib.h>
 
+#include "step_id.h"
 #include "quad.h"
 #include "vertex3d.h"
 #include "contour3d.h"
diff --git a/src/hid/common/vertex3d.c b/src/hid/common/vertex3d.c
index 1704b2b..74c286c 100644
--- a/src/hid/common/vertex3d.c
+++ b/src/hid/common/vertex3d.c
@@ -1,5 +1,7 @@
+#include <glib.h>
 #include <stdlib.h>
 
+#include "step_id.h"
 #include "vertex3d.h"
 
 static int global_vertex3d_count;
diff --git a/src/hid/common/vertex3d.h b/src/hid/common/vertex3d.h
index 3625c5e..8bb6ec9 100644
--- a/src/hid/common/vertex3d.h
+++ b/src/hid/common/vertex3d.h
@@ -6,7 +6,7 @@ typedef struct
   int id;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int vertex_identifier;
+  step_id vertex_identifier;
 } vertex3d;
 
 vertex3d *make_vertex3d (double x, double y, double z);
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 015d575..6e8c5d3 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -15,6 +15,7 @@
 #include "gui-pinout-preview.h"
 #include "pcb-printf.h"
 
+#include "hid/common/step_id.h"
 #include "hid/common/quad.h"
 #include "hid/common/vertex3d.h"
 #include "hid/common/contour3d.h"
diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index 41c287d..5237c4f 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -4,6 +4,7 @@
 
 #include <glib.h>
 
+#include "hid/common/step_id.h"
 #include "hid/common/quad.h"
 #include "hid/common/vertex3d.h"
 #include "hid/common/contour3d.h"
